dipy/core/gradients.py

```diff
@@ -9,11 +9,48 @@
 from dipy.core.geometry import vector_norm, vec2vec_rotmat
 from dipy.core.sphere import disperse_charges, HemiSphere
 
+from dipy.utils.deprecator import deprecate_with_version
+
+
 WATER_GYROMAGNETIC_RATIO = 267.513e6  # 1/(sT)
 
 logger = logging.getLogger(__name__)
 
 
+@deprecate_with_version("dipy.core.gradients.unique_bvals is deprecated, "
+                        "Please use "
+                        "dipy.core.gradients.unique_bvals_magnitude instead",
+                        since='1.2', until='1.4')
+def unique_bvals(bvals, bmag=None, rbvals=False):
```

Comment:
I feel like the changes in this module could be greatly simplified by adding a `tol` key-word argument to this function. If this key-word argument is not set (defaults to None), the function behaves as before. If `tol` is set, it takes precedence.

Reply:
So I should copy my `unique_bvals_tol` into the old `unique_bvals` and use a big `if else`?

Reply:
To smooth the transition, can we do a mix depreciation? 1) Do your suggestion @arokem 2) but warn the user its gonna be split in 2 in Dipy 1.4? I really feel like functions doing two atomic, easy to understand, things could just be 2 functions.

Also if the decision is made to do a if/else approach in that function, @karanphil I would keep the two other, and in the unique_bvals() function, I would do it robust style. If it was my decision (which it is not, I know) I would set both params to None, the user must explicitly choose the behavior (mag or tol) by setting the parameters (likely an integer) and cannot set both.

I say this because I saw a lot of functions with mixed behavior when working on the stateful tractogram and one of the main sources of confusion was default parameters were leading to behavior # 1 but optional parameters were leading to behavior # 2. This is why I like the concept of having two functions better, short term some user will get a warning, long term everything is clearer and no functionalities is lost.

---------------
```
deprication_warning('This function will be split in two', since='1.2', until='1.4')
def unique_bvals(bvals, mag=None, tol=None rbvals=False):
    if mag is None and tol is None:
        raise
    if mag is not None and tol is not None:
         raise
    if mag is not None:
        unique_bvals_mag()
    elif tol is not None:
        unique_bvals_tol()
```